### DISCUSSION TITLE
How do I type hint a question that returns a zip object?

### DISCUSSION THREAD

**Discussion by branlabs** [python/typing#1279] [Q&A]
I've got a function that takes an arbitrary amount of lists (or any iterables, for that matter) and sorts them as one. The code looks like this:
![image](https://user-images.githubusercontent.com/81892257/199738137-1d19559b-97d2-4756-921a-fedd6e71f9db.png)
How can I accurately type hint the function output?

**Comment by jeremyn** [comment]
I did a Google search for this and found [this StackOverflow post](https://stackoverflow.com/q/73202494) which has your exact code in it. (This explains why you pasted an image instead of the code as text.) That post has an [accepted answer](https://stackoverflow.com/a/73202626):

```python
def sort_as_one(*args: T) -> typing.Iterator[T]:
    return zip(*sorted(zip(*args)))
```

Not to make this weird but you should just be up front if you're asking about code from a StackOverflow post. There's nothing wrong with doing that. Is there something else you wanted to ask about it?

**Reply by tankorsmash** [reply]
It's the literal text content of the SO question as well.
